File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        List<Character> vowelList = new ArrayList<>();
        String vowels = "aeiouAEIOU";
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (vowels.indexOf(ch) != -1) {
                vowelList.add(ch);
            }
        }

        Collections.reverse(vowelList);
        int vowelIndex = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (vowels.indexOf(ch) != -1) {
                result.append(vowelList.get(vowelIndex));
                vowelIndex++;
            } else {
                result.append(ch);
            }
        }

        return result.toString();
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
       
        newStr = ""
        vowels = "aeiouAEIOU"
        vowel_list = []

        for char in s:
            if char in vowels:
                vowel_list.append(char)

        vowel_list.reverse()

        for char in s:
            if(char in vowels):
                newStr += vowel_list.pop(0)
            else:
                newStr += char

        return newStr
